x-pack/plugins/ml/public/application/jobs/jobs_list/components/confirm_modals/start_datafeeds_confirm_modal.tsx

```diff
@@ -0,0 +1,142 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect, useCallback, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiSpacer,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiText,
+} from '@elastic/eui';
+import { MlSummaryJob } from '../../../../../../common/types/anomaly_detection_jobs';
+import { isManagedJob } from '../../../jobs_utils';
+// @ts-ignore
+import { startDatafeeds } from '../utils';
+
+type ShowFunc = (jobs: MlSummaryJob[]) => void;
+
+interface Props {
+  setShowFunction(showFunc: ShowFunc): void;
+  unsetShowFunction(): void;
+  refreshJobs(): void;
+  showStartDatafeedModal(job: MlSummaryJob[]): void;
+}
+
+export const StartDatafeedsConfirmModal: FC<Props> = ({
+  setShowFunction,
+  unsetShowFunction,
+  refreshJobs,
+  showStartDatafeedModal,
+}) => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [isManaged, setHasManaged] = useState(true);
+  const [jobsToStart, setJobsToStart] = useState<MlSummaryJob[]>([]);
+
+  const jobIds = useMemo(() => jobsToStart.map(({ id }) => id), [jobsToStart]);
+
+  useEffect(() => {
+    if (typeof setShowFunction === 'function') {
+      setShowFunction(showModal);
+    }
+    return () => {
+      if (typeof unsetShowFunction === 'function') {
+        unsetShowFunction();
+      }
+    };
+  }, []);
+
+  const showModal = useCallback((jobs: MlSummaryJob[]) => {
+    setJobsToStart(jobs);
+
+    if (jobs.some((j) => isManagedJob(j))) {
+      setModalVisible(true);
+      setHasManaged(true);
+    }
+  }, []);
+
+  const closeModal = useCallback(() => {
+    setModalVisible(false);
+    setHasManaged(false);
+  }, []);
+
+  if (modalVisible === false) {
+    return null;
+  }
+
+  if (isManaged) {
+    const title = (
+      <FormattedMessage
+        id="xpack.ml.jobsList.startDatafeedsModal.startDatafeedsTitle"
+        defaultMessage="Start datafeed for {jobsCount, plural, one {{jobId}} other {# jobs}}?"
+        values={{
+          jobsCount: jobIds.length,
+          jobId: jobIds[0],
+        }}
+      />
+    );
+
+    return (
+      <EuiModal data-test-subj="mlStartDatafeedsConfirmModal" onClose={closeModal}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          <p>
+            <>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.ml.jobsList.startDatafeedsModal.startManagedDatafeedsDescription"
+                  defaultMessage="{jobsCount, plural, one {This preconfigured job} other {At least one of these jobs}} is provided by Elastic; starting {jobsCount, plural, one {it} other {them}} might impact other parts of the product."
```

Comment:
Not the end of the world, but if we want "preconfigured" to appear in both singular and plural cases:

>defaultMessage="{jobsCount, plural, one {This preconfigured job is} other {At least one of these jobs is preconfigured and}} provided by Elastic; starting {jobsCount, plural, one {it} other {them}} might impact other parts of the product."

Alternatively, could shorten even more to:


>defaultMessage="{jobsCount, plural, one {This job} other {At least one of these jobs}} is preconfigured by Elastic; starting {jobsCount, plural, one {it} other {them}} might impact other parts of the product."

Reply:
Updated here https://github.com/elastic/kibana/pull/122305/commits/3f17443739229f81b67bfc177d44e83e2e171167